#ifndef SERVERPANELCLIENTLOGINWINDOW_H

///////////////////////////////////////////////////////////////////////////////
/// Definitions //////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

#define SERVERPANELCLIENTLOGINWINDOW_H

///////////////////////////////////////////////////////////////////////////////
/// Headers //////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

#include <QDialog>
#include <QDialogButtonBox>
#include <QLabel>
#include <QLineEdit>
#include <QListWidget>
#include <QPushButton>
#include <QToolButton>
#include <QGridLayout>
#include <QtCrypto>
#include <ServerPanel.h>
#include <ServerPanelClientMainWindow.h>

///////////////////////////////////////////////////////////////////////////////
/// Namespace ////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////



///////////////////////////////////////////////////////////////////////////////
/// ServerPanelClientLoginWindow Class Definition ////////////////////////////
/////////////////////////////////////////////////////////////////////////////

class ServerPanelClientLoginWindow : public QDialog {
    // Make sure this is an instance of a QObject
    Q_OBJECT
// Public
public:
    // Singleton
    static   ServerPanelClientLoginWindow* Instance(QWidget* cParent = 0);
    // Constructor
    explicit ServerPanelClientLoginWindow          (QWidget* cParent = 0);
    // Destructor
            ~ServerPanelClientLoginWindow          ();
// Protected
protected:
    // Properties
    static ServerPanelClientLoginWindow* mInstance;
    QGridLayout*                         mGrid;
    // Methods
    void SetupButtons                              ();
    void SetupLineEdits                            ();
    void SetupLists                                ();
// Protected slots
protected slots:

};

#endif // SERVERPANELCLIENTLOGINWINDOW_H
